app/src/main/java/de/westnordost/streetcomplete/quests/opening_hours/ResurveyOpeningHours.kt

```diff
@@ -0,0 +1,78 @@
+package de.westnordost.streetcomplete.quests.opening_hours
+
+import android.util.Log
+import de.westnordost.osmapi.map.data.BoundingBox
+import de.westnordost.osmapi.map.data.Element
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.meta.OsmTaggings
+import de.westnordost.streetcomplete.data.osm.OsmElementQuestType
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.download.MapDataWithGeometryHandler
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao
+import de.westnordost.streetcomplete.data.osm.tql.getQuestPrintStatement
+import de.westnordost.streetcomplete.data.osm.tql.toGlobalOverpassBBox
+import de.westnordost.streetcomplete.quests.DateUtil
+
+class ResurveyOpeningHours(private val overpassServer: OverpassMapDataDao) : OsmElementQuestType<OpeningHoursAnswer> {
+    override val commitMessage = "resurvey opening hours"
+    override val icon = R.drawable.ic_quest_opening_hours
+    override fun getTitle(tags: Map<String, String>) = R.string.resurvey_opening_hours_title
+    override fun getTitleArgs(tags: Map<String, String>, featureName: Lazy<String?>): Array<String> {
+        return arrayOf(tags["name"]!!)
+    }
+
+    override fun download(bbox: BoundingBox, handler: MapDataWithGeometryHandler): Boolean {
+        return overpassServer.getAndHandleQuota(getOverpassQuery(bbox)) { element, geometry ->
+            if (element.tags != null) {
+                // require opening hours that are supported
+                if (OpeningHoursTagParser.parse(element.tags["opening_hours"]!!) != null) {
+                    handler.handle(element, geometry)
+                } else {
+                    Log.wtf("AAAA", "opening_hours=" + element.tags["opening_hours"] + " was rejected as not representable in SC")
```

Comment:
This should go before this is released

Reply:
`Log.wtf("AAAA"` should certainly go away and be fixed, but I think that some way of logging "some oh quests were discarded as syntax was unsupported" would be useful.

Though I have no strong opinion here.

Reply:
Only us developers are going to see this log.

If we want reliable statistics, better would be a (global) analysis of how much % of opening hours use which syntax feature. Should be interesting. One could use the OpeningHoursParser library in a small java application that downloads all the opening hours (in the world) and counts features. Maybe Simon already did that.

Reply:
I was thinking rather about making debugging "why this quest is not generated" easier. More than once I though that  found some problem only to discover on looking into logs that it is 2000 object limit, or note on the element or something similar.

Reply:
> One could use the OpeningHoursParser library in a small java application that downloads all the opening hours (in the world) and counts features. Maybe Simon already did that.

Done already, but some time ago :) See https://github.com/simonpoole/OpeningHoursParser#openinghoursparser

> It parses 146'825 (91%) of 161'268 unique test strings in non-strict mode. The remaining 14'443 are likely valid errors, spot checking shows that they have obvious issues. In strict mode further 22'440 fail (total 36'883).

I also run a similar code, but I was looking only for crashes to ensure that this code will not crash on weird tags. And even found something, see https://github.com/simonpoole/OpeningHoursParser/issues/26 . It was done as a monstrous hack - see https://github.com/matkoniecz/Zazolc/commit/7245406f9538c9a880a0b8805df348f347f21baa

Reply:
Oh, so #26 was never fixed and you had to workaround it?

Reply:
#26 seems unrelated - it is about a wheelchair access quest

Reply:
I meant https://github.com/simonpoole/OpeningHoursParser/issues/26

Reply:
Was fixed in 17.1 - see https://github.com/simonpoole/OpeningHoursParser/releases